### DISCUSSION TITLE
Use subworkflow from a relative path in the same repository

### DISCUSSION THREAD

**Discussion by gustaff-weldon** [actions/runner#1419] [Q&A] [answered]
Is there a way to use sub-workflow which lives in the same repository and reference it via a relative path?

We want to use the version which lives in the commit that triggered the build (as it could have been modified in the very same commit...)

We can use composite actions like this:
```
    steps:
      - name: Log in to Amazon ECR
        uses: ./.github/actions/amazon-ecr-login
```

We want the same with workflows:
```
jobs:
  api-docs:
    uses: ./.github/workflows/api-docs.yml
```

Docs do not mention it https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_iduses, and, unsurprisingly, the above does not work.

An attempt  to force version to current sha, did not work either (`{{ github.sha }}` was not interpolated):
```
jobs:
  api-docs:
    uses: myorg/myrepo/.github/workflows/api-docs.yml@${{ github.sha }}
```

Any ideas? We were very excited about the ability to split our humongous monorepo CI yaml into subworkflow files, but that inability to just use the file that lives next to the main workflow looks like a dealbreaker...

**Comment by jenschelkopf** [comment]
Hi @gustaff-weldon, I'm the product manager for reusable workflows. Thanks for trying out the feature and letting us know your thoughts so far.

You mentioned that this is a dealbreaker for you. Can you expand on that? Why does it stop you from breaking up your humongous monorepo?

**Reply by gustaff-weldon** [reply]
hi @jenschelkopf, thanks for getting back. Let me give you more background on what we are trying to achieve.

### Current state
We are running a private monorepo with GH actions setup for CI. Our main workflow file, `ci.yml` is over 2400 lines long. That is way too much. 
```
.github
├── CODEOWNERS
├── actions
│   ├── amazon-ecr-login
│   │   ├── README.md
│   │   └── action.yml
...
└── workflows
    └── ci.yml
```

#### Reuse composite actions
With the recent upgrade of composite actions we were able to extract repeated steps into those. 

We reference our composite actions using **relative path** (as shown in https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-using-an-action-in-the-same-repository-as-the-workflow):
```
   steps:
      - name: Log in to Amazon ECR
        uses: ./.github/actions/amazon-ecr-login
```

This works great.

### Expected workflow usage
We want the same to be possible with subworkflows. We would like to be able to reference our sub workflows using **relative path**:

```
└── workflows
    ├── ci.yml
    ├── frontend.yml
    └── api-docs.yml
```
and in `ci.yml`:

```
jobs:
  api-docs:
    uses: ./.github/workflows/api-docs.yml
```

This would allow us to update the workflow file and see the outcome of changes in the build of that PR. 

### Why current approach with `{owner}/{repo}/{path}/{filename}@{ref}` is cumbersome for our use case

Our main goal is to split our main `ci.yml` into subworkflows. We might reuse some of them, but that goes second.

When we update our CI, we want to be able to test changes immediately in the PR with the changes.
Referencing by version forces usage of SHA, which would introduce an awkward procedure:
- make changes to subworkflow
- commit, take note of sha
- update reference to the workflow in main `ci.yml` with the above sha
- push to branch

This will break, if, eg. you rebase your branch. It's very fragile and prone to errors, thus we would not like to go for it, hence a dealbreaker.

**Reply by jenschelkopf** [reply]
Okay, got it. Thanks for the feedback

**Reply by gustaff-weldon** [reply]
You are welcome. Out of curiosity (not really, cause I would really like to see this implemented ;)): given that relative paths already work for composite actions, I wonder how applicable will be their solution in workflows? Do you see a chance for this to be added in a foreseeable future?

**Comment by AllanOricil** [comment]
Is this released?

**Comment by asvetlov** [comment]
Have the same issue (splitting ci.yml into subworklows): https://github.com/aio-libs/aiohttp/pull/6264

Would be nice to have relative workflow path support!

**Reply by AllanOricil** [reply]
seems weird that they did not think about that

**Comment by hubol** [comment]
Would love this! Have similar motivations as the author.

**Comment by solarmosaic-kflorence** [comment]
See https://github.community/t/ref-head-in-reusable-workflows/203690

**Comment by chehsunliu** [comment]
For those people who uses GitHub Enterprise, this feature (or bug fix?) is released in `v3.5.0`: https://docs.github.com/en/enterprise-server@3.5/admin/release-notes#3.5.0.rc1.

**Comment by hubol** [comment]
Hey! This seems to work now: https://github.blog/changelog/2022-01-25-github-actions-reusable-workflows-can-be-referenced-locally/
```yml
jobs:
  call-workflow-in-local-repo:
    uses: ./.github/workflows/workflow-2.yml
```

For me, I had to use the ./ at the beginning instead of my organization name.

**Comment by gustaff-weldon** [comment, marked as answer]
@hubol yes, Github released this feature: https://github.com/github/roadmap/issues/98 and https://docs.github.com/en/actions/using-workflows/reusing-workflows

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
